R/mcmc.R

```diff
@@ -136,29 +160,36 @@ split_dim <- function(a, n) {
 }
 
 
-#' Title - TODO
+#' Extract draws from a `stanfit` object
+#'
+#' @description
+#' Extract draws from a `stanfit` object, manipulate it
+#' and combine them together into a list.
+#'
+#' @param stan_fit A `stanfit` object.
+#'
+#' @return
+#' A named list of length 2 containing:
+#' - `beta`: a list of length equal to the number of draws containing
+#'   the draws from the posterior distribution of the regression coefficients.
+#' - `sigma`: a list of length equal to the number of draws containing
+#'   the draws from the posterior distribution of the covariance matrices. Each element
+#'   of the list is a list with length equal to 1 if `same_cov = TRUE` or equal to the
+#'   number of groups if `same_cov = FALSE`.
 #'
-#' @param stan_fit TODO
 #' @importFrom rstan extract
 extract_draws <- function(stan_fit) {
 
     pars <- extract(stan_fit, pars = c("beta", "Sigma"))
     names(pars) <- c("beta", "sigma")
 
     ##################### from array to list
-    pars$sigma <- split_dim(pars$sigma, 1)
+    pars$sigma <- split_dim(pars$sigma, 1) # list of length equal to the number of draws
 
-    if (length(dim(pars$sigma[[1]])) == 3) { # if same_cov == FALSE
-        pars$sigma <- lapply(
-            pars$sigma,
-            function(x) split_dim(x, 1)
-        )
-    } else {
```

Comment:
Looks like you've collapsed the logic for same_cov  =TRUe / FALSe into a single block, do you have unit tests that show it work in both scenarios ?

Reply:
Yes, the tests I have added in test-mcmc regards exactly this. Essentially the "else" was never ran since the if condition is always true.